python/ray/serve/scripts.py

```diff
@@ -563,17 +563,24 @@ def run(
                 yield_on_timeout=True,
             ):
                 if changes:
-                    cli_logger.info(
-                        f"Detected file change in path {watch_dir}. Redeploying app."
-                    )
-                    # The module needs to be reloaded with `importlib` in order to pick
-                    # up any changes.
-                    app = _private_api.call_app_builder_with_args_if_necessary(
-                        import_attr(import_path, reload_module=True), args_dict
-                    )
-                    serve.run(
-                        target=app, blocking=True, name=name, route_prefix=route_prefix
```

Comment:
Also note there is a bug in here when previously introduced the blocking function. The reload should be unblocked so file watching can continuously redeploy. This regression will be caught by the newly introduced test in this PR